src/changeKeys.js

```diff
@@ -4,18 +4,29 @@ import pressedAnotherOperator from "./pressedAnotherOperator";
 const changeKeys = (result, button) => {
   const lastCharIsNum = "0123456789".includes(result[result.length - 1]);
   const lastCharIsOperator = "/*+-".includes(result[result.length - 1]);
+  const isOperator = "/*+-".includes(button);
 
   if (result === "") {
-    if ("+-(0123456789".includes(button)) {
+    if ("(0123456789".includes(button)) {
       return button;
     }
+    if (isOperator) {
+      return (result = "0" + button);
+    }
   }
   if (lastCharIsNum) {
     return ([result, button] = lastCharNumber(result, button));
   }
 
   if (lastCharIsOperator) {
-    return ([result, button] = pressedAnotherOperator(result, button));
+    if (".".includes(button)) {
+      return result + "0.";
+    }
+
+    if (pressedAnotherOperator) {
+      result = result.slice(0, -1);
+      return [result, button];
```

Comment:
@grimxreaper what's different about this return from the others?

Reply:
It is returning two values

Reply:
What is the data type returned here and the data type returned elsewhere?

Reply:
Here it is a destructuring assignment, returning an array, and elsewhere it is just a returned variable...

Reply:
So it returns an array here. Cool. And what is the type of the variable returned elsewhere?

Reply:
What kinds of issues could occur from returning an array here and a variable in other places?

Reply:
> So it returns an array here. Cool. And what is the type of the variable returned elsewhere?

The type of the variables returned elsewhere are numbers.

> What kinds of issues could occur from returning an array here and a variable in other places?

Ahh...how I then manipulate what is returned would have to be altered to cater to the _**type**_ of the returned value. Also, how I reference what is returned.
For example, in the commit 'simplify pressedAnotherOperator' I had to make an alteration to `hasLastChar` so that it would fix a bug that could not slice or use substr on a number, which was the type of what was returned, so it had to be made into a string first.

`  const hasLastChar = (chars) => {
    return (result + "").substr(-1) === chars;
  };`

Reply:
Yes, if you return different types of things from the same function, you have to handle every type of thing that could come back when you use it. It’s a good idea to make sure your function returns the same type of thing so you don’t have to look inside of it when you use it.

Reply:
Writing down all your great feedback.